process and unit-tests

LeKaitoW opened the pull request.

LeKaitoW committed:
initial process
add process blocks
add unit test

LeKaitoW committed:
use docks instead of input/output blocks

LeKaitoW committed:
generate situation in test instead of process constructor

LeKaitoW committed:
update test

LeKaitoW committed:
add queue block
update tests

LeKaitoW committed:
add test block
update tests

LeKaitoW committed:
code claen up

AlexChernov committed:
Добавлена подержка junit тестов

AlexChernov committed:
Добавлен junit тест для ru.bmstu.rk9.rao.lib.math

AlexChernov committed:
Добавлен junit тест для ru.bmstu.rk9.rao.ui.trace.Tracer

LeKaitoW committed:
process code clean up
update OutputDocks (aurusov/raox#285)

AlexChernov committed:
Добавил exclude Unit тестов из integration-test фазы

LeKaitoW committed:
add config

LeKaitoW committed:
update config

LeKaitoW committed:
add target

LeKaitoW committed:
move unit tests

LeKaitoW committed:
update packages

LeKaitoW committed:
remove unused

LeKaitoW committed:
remove unused linkedDock

aurusov commented on ru.bmstu.rk9.rao.tests/pom.xml:
```diff
@@ -14,9 +14,69 @@
+						<sources>
+							<source>${basedir}/src/ru/bmstu/rk9/rao/tests</source>
+						</sources>
+						</configuration>
```
Табуляция

aurusov commented on ru.bmstu.rk9.rao.tests/pom.xml:
```diff
@@ -14,9 +14,69 @@
+						<phase>test</phase>
+						<goals>
+							<goal>test</goal>
+						</goals>
```
1. Если мы сами для себя называем эти тесты unit-тесты, то и слово test надо переименовать в unit-test везде где можно
2. Добавить директорию, из которой берутся unit-тесты

aurusov commented on ru.bmstu.rk9.rao.tests/pom.xml:
```diff
@@ -25,6 +85,9 @@
 				<configuration>
+					<excludes>
+						<exclude>**/unit/**</exclude>
+					</excludes>
```
@AlexChernov 
В тестах на Дженкинсе видно три стадии `T E S T S`. Как они называются, для чего нужны ?

aurusov replied:
Поговорили, решили
- убрать плагин для интеграционного тестирования
- убрать этот `exclude`

AlexChernov committed:
Плагин для интеграционного тестирования отключен, убрал exclude
У плагина maven-surefire-plugin id переименовано на unit-test
Добавлена директорию, из которой берутся unit-тесты

AlexChernov committed:
Исправлен запуск unit тестов процессного подхода

AlexChernov committed:
Исправлена табуляция

AlexChernov committed:
Исправлен path style

LeKaitoW committed:
fix warnings
